### DISCUSSION TITLE
Vertext buffer exceeding it's size

### DISCUSSION THREAD

**Discussion by moong23** [visgl/deck.gl#8864] [Q&A] [answered]
I'm trying to create a PathLayer.
But I'm struggling with some rendering error rn.

### error log
`[.WebGL-0x12800c53400] GL_INVALID_OPERATION: Vertex buffer is not big enough for the draw call`

can't really put real-data in here but my data has this format
```jsx
data : 
[
  {
    id: 'idstring',
    pathData: //Array(24)
      [
        {
          [126.xxxxxxx, 37.xxxxxxxx],...
        }, 
        ...
      ],
      [
        ....
      ],
      ...
  },
  ...
]
```
which makes 6 * Array(24) * Array(60) * Array(2) points in the data.
(6 : # of trajectories / 24 : # of trajectories in one path / 60 : # of points in one trajectory / 2 : [number, number]

and I have several (maybe upto 50) trajectories i have to render.

I know I can make multiple pathlayers with trajectorys as 1:1 ratio, but I want to know if it is possible to put all the trajecotories in one path layer.

btw, the code i'm using to render pathLayer looks like this

```jsx
new PathLayer({
  id: 'trajLayer',
  data: data,
  ...,
  getPath: (d:any) => d.pathData
})
```

**Comment by Pessimistress** [comment, marked as answer]
`number[][][]` is not a valid path. See https://deck.gl/docs/api-reference/layers/path-layer#getpath

**Reply by moong23** [reply]
opps forgot about the data format :(
thx for the help!
